fragment.go

```diff
@@ -1997,3 +2021,67 @@ func byteSlicesEqual(a [][]byte) bool {
 func pos(rowID, columnID uint64) uint64 {
 	return (rowID * ShardWidth) + (columnID % ShardWidth)
 }
+
+// vector stores the mapping of colID to rowID.
+// It's used for a mutex field type.
+type vector interface {
+	Get(colID uint64) (uint64, bool)
+	Set(colID, rowID uint64)
+}
+
+// mapVector implements the vector interface using a map.
+type mapVector struct {
+	mu sync.RWMutex
+	m  map[uint64]uint64
+}
+
+// newMapVector returns a mapVector.
+func newMapVector() *mapVector {
```

Comment:
let's either delete mapVector or add a comment on it as to why it's unused but still around